### DISCUSSION TITLE
relationship between contact model acceleration term 'a_0` and classical acceleration

### DISCUSSION THREAD

**Discussion by bilal-bdai** [loco-3d/crocoddyl#1323] [Q&A]
for writing the contact model we have 
$$
J \ddot{q} + \dot{J} \dot{q} = a_{spatial} 
$$
what's the explanation for writing $a^{classical}_{linear} = \dot{J}_{linear} \dot{q}$  [here](https://github.com/loco-3d/crocoddyl/blob/master/include/crocoddyl/multibody/contacts/contact-3d.hxx#L76) 

I tried to write it down but i can't seem to make the connection

**Comment by cmastalli** [comment]
Hi @bilal-bdai,

This constraint imposes contact accelerations equal to zero. However, spatial acceleration is a property of a body as a whole, not of a body-fixed point. Indeed, we are interested in constraining the classical acceleration, not the spatial acceleration. Moreover, we are considering its linear part only as this is a 3D contact model.

Note that we also include the Baumgarte stabilization terms in `a0`.
